### PR TITLE
report: let fireworks eligibility ignore PWA category

### PR DESCRIPTION
**Pull request by paulirish**
IMO this guy deserved it: https://twitter.com/peruvianidol/status/1288839520272683009

### CODE REVIEW

File: lighthouse-core/report/html/renderer/report-ui-features.js [hunk 1]
```diff
@@ -98,11 +98,15 @@ class ReportUIFeatures {
...
-    const scoresAll100 = Object.values(report.categories).every(cat => cat.score === 1);
-    const hasAllCoreCategories =
-      Object.keys(report.categories).filter(id => !Util.isPluginCategory(id)).length >= 5;
-    if (scoresAll100 && hasAllCoreCategories) {
+    // Fireworks!
+    // To get fireworks you need 100 scores in all categories…
+    // except plugins and PWA (because going the PWA route is discretionary).
+    const fireworksEligibleCategories = Object.values(report.categories)
+      .filter(cat => !Util.isPluginCategory(cat.id))
+      .filter(cat => cat.id != 'pwa');
```
Comment (patrickhulce) [inline lighthouse-core/report/html/renderer/report-ui-features.js:106]:
```suggestion
      .filter(cat => cat.id !== 'pwa');
```

Comment (brendankenny) [inline lighthouse-core/report/html/renderer/report-ui-features.js:106]: should have a test

File: lighthouse-core/report/html/renderer/report-ui-features.js [hunk 2]
```diff
@@ -98,11 +98,15 @@ class ReportUIFeatures {
...
 
-    // Fireworks.
-    const scoresAll100 = Object.values(report.categories).every(cat => cat.score === 1);
-    const hasAllCoreCategories =
-      Object.keys(report.categories).filter(id => !Util.isPluginCategory(id)).length >= 5;
-    if (scoresAll100 && hasAllCoreCategories) {
+    // Fireworks!
+    // To get fireworks you need 100 scores in all categories…
+    // except plugins and PWA (because going the PWA route is discretionary).
+    const fireworksEligibleCategories = Object.values(report.categories)
```
Comment (brendankenny) [inline lighthouse-core/report/html/renderer/report-ui-features.js:104]:
```suggestion
    const fireworksRequiredCategories = Object.values(report.categories)
```
?

It's sort of both, but treated below as the requirements when checking the scores and count below

File: lighthouse-core/report/html/renderer/report-ui-features.js [hunk 3]
```diff
@@ -98,11 +98,15 @@ class ReportUIFeatures {
...
+      .filter(cat => !Util.isPluginCategory(cat.id))
+      .filter(cat => cat.id != 'pwa');
+    const scoresAll100 = fireworksEligibleCategories.every(cat => cat.score === 1);
+    const doesRunIncludeCoreCategories = fireworksEligibleCategories.length === 4;
```
Comment (brendankenny) [inline lighthouse-core/report/html/renderer/report-ui-features.js:108]: If we're going to say "exactly four", let's just test the four by name? It's explicit, the core ids haven't changed in ages, it should trigger the existing unit tests in a clear way if a new category is added to `default-config`, and it doesn't give a free fireworks show to folks using a custom config with their own categories.

Comment (paulirish) [reply lighthouse-core/report/html/renderer/report-ui-features.js:108]: yeah, had been considering the same thing.  ok, sold.

**Comment by connorjclark** [comment]
@paulirish seems you wanted to make some more modifications?

**Pull request merged**